src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -75,7 +75,11 @@ public override void OnEnd(Activity activity)
 
             if (snapshot != pointers)
             {
+                // TODO: Remove this when SetStatus is deprecated
                 activity.SetStatus(Status.Error);
```

Comment:
I assume there's never a time we'd call this version of `SetStatus` and not also want to call the "real" version, right?

Here's another spot

https://github.com/open-telemetry/opentelemetry-dotnet/blob/a274f06b2c4c3e76533695b0bd31e177c86b1a30/src/OpenTelemetry.Shims.OpenTracing/SpanBuilderShim.cs#L176

We could kill two stones with one bird if we moved the call to `activity.SetStatus(ActivityStatusCode)` from our extension method. Though, from what I can tell I think that OpenTracing shim is the only other place... so maybe easier to just duplicate the call there as well.

Reply:
@alanwest I was not aware of this one, will get it updated. what do you think about marking our extension method obsolete?

Reply:
I would support marking it obsolete.

Reply:
+1